stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/usecase/LoadMessageByIdImpl.kt

```diff
@@ -0,0 +1,42 @@
+package io.getstream.chat.android.offline.usecase
+
+import androidx.annotation.CheckResult
+import io.getstream.chat.android.client.call.Call
+import io.getstream.chat.android.client.call.CoroutineCall
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.livedata.utils.validateCid
+import io.getstream.chat.android.offline.ChatDomainImpl
+
+public interface LoadMessageById {
+    /**
+     * Loads message for a given message id and channel id
+     *
+     * @param cid: the full channel id i. e. messaging:123
+     * @param messageId: the id of the message
+     * @param olderMessagesOffset: how many new messages to load before the requested message
+     * @param newerMessagesOffset: how many new messages to load after the requested message
+     */
+    @CheckResult
+    public operator fun invoke(
+        cid: String,
+        messageId: String,
+        olderMessagesOffset: Int = 0,
+        newerMessagesOffset: Int = 0,
+    ): Call<Message>
+}
+
+internal class LoadMessageByIdImpl(private val domainImpl: ChatDomainImpl) : LoadMessageById {
```

Comment:
Why we need to duplicate `LoadMessageByIdImpl`

Reply:
The same reason. It's interface from the `livedata` package is public and we must keep it